### PR TITLE
EE-1086: add InvalidPublicKey variant to auction Error and use it

### PR DESCRIPTION
**Pull request by henrytill**
Ref: https://casperlabs.atlassian.net/browse/EE-1086

### CODE REVIEW

**Comment by henrytill** [comment]
**NOTE:** 
I also corrected a few places where we were using in `InvalidContext` instead of `InvalidCaller`.

File: types/src/auction.rs [hunk 1]
```diff
@@ -339,7 +339,7 @@ pub trait Auction:
     /// Accessed by: node
     fn run_auction(&mut self) -> Result<()> {
         if self.get_caller() != SYSTEM_ACCOUNT {
```
Comment (mpapierski) [inline types/src/auction.rs:341]: According to Display for Error shouldn't those 2 cases be still `InvalidCaller?` 

> #[fail(display = "Function must be called by system account")]
> InvalidCaller = 16,

**Review by mpapierski — APPROVED** [review]
LGTM nevermind my deleted comment

**Pull request merged**